modules/ivc_champva/spec/requests/ivc_champva/v1/forms/ves_uploads_spec.rb

```diff
@@ -28,136 +29,144 @@
+    allow(mock_s3).to receive(:put_object).and_return({ success: true })
   end
 
   after do
     Aws.config = @original_aws_config
   end
 
-  describe '#submit with VES integration' do
-    let(:form_data) do
-      JSON.parse(Rails.root.join('modules', 'ivc_champva', 'spec', 'fixtures', 'form_json', 'vha_10_10d.json').read)
-    end
-
-    context 'with flipper champva_send_to_ves enabled' do
-      before do
-        allow(Flipper).to receive(:enabled?)
-          .with(:champva_send_to_ves, anything)
-          .and_return(true)
-      end
-
-      context 'when environment is not production' do
-        before do
-          allow(Settings).to receive(:vsp_environment).and_return('staging')
-        end
-
-        it 'uploads a PDF file to S3 and submits to VES for form 10-10D' do
-          post '/ivc_champva/v1/forms', params: form_data
-
-          record = IvcChampvaForm.first
-          expect(record.first_name).to eq('Veteran')
-          expect(record.last_name).to eq('Surname')
-          expect(record.form_uuid).to be_present
-
-          expect(IvcChampva::VesDataFormatter).to have_received(:format_for_request)
-          expect(ves_client).to have_received(:submit_1010d).with('fake-id', 'fake-user', ves_request)
-          expect(mock_form).to have_received(:update).with(
-            hash_including(
-              application_uuid: 'test-uuid',
-              ves_status: 200
-            )
-          )
-
-          expect(response).to have_http_status(:ok)
-        end
-
-        it 'handles VES formatter errors gracefully' do
-          allow(IvcChampva::VesDataFormatter).to receive(:format_for_request)
-            .and_raise(StandardError.new('formatting error'))
-
-          post '/ivc_champva/v1/forms', params: form_data
-
-          expect(response).to have_http_status(:internal_server_error)
-          expect(response.parsed_body['error_message']).to eq('Error: formatting error')
-        end
-
-        it 'handles nil VES request gracefully' do
-          allow(IvcChampva::VesDataFormatter).to receive(:format_for_request).and_return(nil)
-
-          post '/ivc_champva/v1/forms', params: form_data
-
-          expect(response).to have_http_status(:internal_server_error)
-          expect(response.parsed_body['error_message']).to eq('Error: Failed to format data for VES submission')
-        end
-
-        it 'handles VES API errors gracefully and still returns success' do
-          allow(ves_client).to receive(:submit_1010d).and_raise(StandardError.new('api error'))
-
-          post '/ivc_champva/v1/forms', params: form_data
-
-          # Should still be successful even if VES fails
-          expect(response).to have_http_status(:ok)
+  describe 'run this section with both values of champva_retry_logic_refactor expecting identical behavior' do
+    retry_logic_refactor_values = [true, false]
+    retry_logic_refactor_values.each do |champva_retry_logic_refactor_state|
```

Comment:
Added code to iterate through both values for `champva_retry_logic_refactor_state` and assign them to the `champva_retry_logic_refactor` toggle for this block of tests.